gateway/app/main/events/fl_events.py

```diff
@@ -0,0 +1,153 @@
+import uuid
+import json
+
+from .socket_handler import SocketHandler
+from ..codes import MSG_FIELD, RESPONSE_MSG, CYCLE
+from ..processes import processes
+
+# Singleton socket handler
+handler = SocketHandler()
+
+
+def host_federated_training(message: dict, socket) -> str:
+    """This will allow for training cycles to begin on end-user devices.
+
+        Args:
+            message : Message body sended by some client.
+            socket: Socket descriptor.
+        Returns:
+            response : String response to the client
+    """
+    data = message[MSG_FIELD.DATA]
+    response = {}
+
+    try:
+        # Retrieve JSON values
+        serialized_model = data.get(MSG_FIELD.MODEL, None)
+        serialized_client_plans = data.get(CYCLE.TRAINING_PLANS, None)
+        serialized_client_protocols = data.get(CYCLE.SECURE_AGG_PROTOCOL, None)
+        serialized_avg_plan = data.get(CYCLE.AVG_PLAN, None)
+        client_config = data.get(CYCLE.CLIENT_CONFIG, None)
+        server_config = data.get(CYCLE.SERVER_CONFIG, None)
+
+        # Create a new FL Process
+        fl_process = processes.create_process(
+            model=serialized_model,
+            client_plans=serialized_client_plans,
+            client_protocols=serialized_client_protocols,
+            server_averaging_plan=serialized_avg_plan,
+            client_config=client_config,
+            server_config=server_config,
+        )
+
+    except Exception as e:  # Retrieve exception messages such as missing JSON fields.
+        response[RESPONSE_MSG.ERROR] = str(e)
+
+    return json.dumps(response)
+
+
+def authenticate(message: dict, socket) -> str:
+    """ New workers should receive a unique worker ID after authenticate on PyGrid platform.
+        
+        Args:
+            message : Message body sended by some client.
+            socket: Socket descriptor.
+        Returns:
+            response : String response to the client
+    """
+    response = {}
+
+    # Create a new worker instance and bind it with the socket connection.
+    try:
+        # Create new worker id
+        worker_id = str(uuid.uuid4())
```

Comment:
This authentication is fine for me, since we really just need a way to recognize the worker somehow. If you guys want to provide something more detailed or secure here, please feel free to use your best judgment. :)